data_model/src/executor.rs

```diff
@@ -81,6 +83,9 @@ mod model {
         ///
         /// These IDs refer to the types in the schema.
         pub permissions: BTreeSet<PermissionId>,
+        /// Type id in the schema.
+        /// Corresponds to payload of `InstructionBox::Custom`.
+        pub custom_instruction_definition_id: Option<Ident>,
```

Comment:
```suggestion
        pub instruction_id: Option<Ident>,
```

Reply:
it's a bit weird that it's an `Option` and not a `Vec`

Reply:
Potentially it is possible to use arbitrary type for payload of custom instruction. If it is enum, in schema there will be `iroha_schema::Metadata::Enum` with vector of variants